core/src/com/mygdx/game/models/ConfigurationFactory.java

```diff
@@ -12,15 +12,15 @@
 
     private ConfigurationFactory(){}
 
-    private HashMap<String, Terrain> terrainHashMap;
```

Comment:
Is there a reason that this can't be Terrain?

Reply:
Otherwise looks fine

Reply:
Upcasting from TerrainClass to Terrain (from class to interface)? Thought it was a typo.

Reply:
Casting from TerrainClass to Terrain (from class to interface)? Thought it was a typo. Reverted the renaming.